extensions-core/lookups-cached-single/src/main/java/io/druid/server/lookup/jdbc/JdbcDataFetcher.java

```diff
@@ -0,0 +1,230 @@
+    this.table = Preconditions.checkNotNull(table, "table");
+    this.keyColumn = Preconditions.checkNotNull(keyColumn, "keyColumn");
+    this.valueColumn = Preconditions.checkNotNull(valueColumn, "valueColumn");
+
+    this.fetchAllQuery = String.format(
+        "SELECT %s, %s FROM %s",
+        this.keyColumn,
+        this.valueColumn,
+        this.table
+    );
+    this.fetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.valueColumn,
+        this.table,
+        this.keyColumn
+    );
+    this.reverseFetchQuery = String.format(
+        "SELECT %s FROM %s WHERE %s = :val",
+        this.keyColumn,
+        this.table,
+        this.valueColumn
+    );
+    dbi = new DBI(
+        connectorConfig.getConnectURI(),
+        connectorConfig.getUser(),
+        connectorConfig.getPassword()
+    );
+    dbi.registerMapper(new KeyValueResultSetMapper(keyColumn, valueColumn));
+  }
+
+  @Override
+  public Iterable<Map.Entry<String, String>> fetchAll()
+  {
+    return inReadOnlyTransaction(new TransactionCallback<List<Map.Entry<String, String>>>()
+                                 {
+                                   @Override
+                                   public List<Map.Entry<String, String>> inTransaction(
+                                       Handle handle,
+                                       TransactionStatus status
+                                   ) throws Exception
+                                   {
+                                     return handle.createQuery(fetchAllQuery)
+                                                  .setFetchSize(streamingFetchSize)
+                                                  .map(new KeyValueResultSetMapper(keyColumn, valueColumn))
+                                                  .list();
+                                   }
+
+                                 }
+    );
+  }
+
+  @Override
+  public String fetch(final String key)
+  {
+    List<String> pairs = inReadOnlyTransaction(
+        new TransactionCallback<List<String>>()
+        {
+          @Override
+          public List<String> inTransaction(Handle handle, TransactionStatus status) throws Exception
+          {
+            return handle.createQuery(fetchQuery)
+                         .bind("val", key)
+                         .map(StringMapper.FIRST)
+                         .list();
+          }
+        }
+    );
+    if (pairs.isEmpty()) {
+      return null;
+    }
+    return Strings.nullToEmpty(pairs.get(0));
+  }
+
+  @Override
+  public Iterable<Map.Entry<String, String>> fetch(final Iterable<String> keys)
+  {
+    QueryKeys queryKeys = dbi.onDemand(QueryKeys.class);
+    return queryKeys.findNamesForIds(Lists.newArrayList(keys), table, keyColumn, valueColumn);
+  }
+
+  @Override
+  public List<String> reverseFetchKeys(final String value)
+  {
+    List<String> results = inReadOnlyTransaction(new TransactionCallback<List<String>>()
```

Comment:
thought this will make it faster ? i am following the  code present at Druid base code.